### ISSUE TITLE
The maven codegen plugin does not load syntheticObjects from external configurationFile

### ISSUE THREAD

**Issue by alf** [jOOQ/jOOQ#13897]
### Expected behavior

I’ve configured `syntheticPrimaryKeys` and `syntheticForeignKeys` in an external xml file according to the xsd and include this file in my maven configuration by using the `<configurationFile>`-tag. I expect this to work the same as if I provide this configuration in the maven configuration, unfortunately this is not so.

### Actual behavior

The problem is that the loaded field-definitions are blank when I move the configuration to the external file. That is to say that given the following xml in the external file, the `MiniJAXB.unmarshal`-kode produces a primaryKey-definition where the `List<String> fields` only contains blank strings.

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<configuration xmlns="http://www.jooq.org/xsd/jooq-codegen-3.16.5.xsd">
    <generator>
        <database>
            <syntheticObjects>
                <primaryKeys>
                    <primaryKey>
                        <name>AKSEPTANSETYPE__PK</name>
                        <tables>KJERNEAPI\.AKSEPTANSETYPE</tables>
                        <fields>
                            <field>INSTITUSJONSNR_EIER</field>
                            <field>AKSEPTANSETYPEKODE</field>
                        </fields>
                    </primaryKey>
...
```

If we set a breakpoint on `MiniJAXB.java:226` with the condition `element.getTagName().equals("field")`, we see that the attempt to deserialize the `<field>`-tags fails because `unmarshal0` does not handle text nodes.

In other words, MiniJAXB does not seem to handle lists of strings properly?

### Versions

- jOOQ: 3.16.9
- Java: 11

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can reproduce this minimally as such:

```java
        println(MiniJAXB.marshal(MiniJAXB.unmarshal(
            """
            <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
            <configuration xmlns="http://www.jooq.org/xsd/jooq-codegen-3.16.5.xsd">
                <generator>
                    <database>
                        <syntheticObjects>
                            <primaryKeys>
                                <primaryKey>
                                    <name>AKSEPTANSETYPE__PK</name>
                                    <tables>KJERNEAPI\\.AKSEPTANSETYPE</tables>
                                    <fields>
                                        <field>INSTITUSJONSNR_EIER</field>
                                        <field>AKSEPTANSETYPEKODE</field>
                                    </fields>
                                </primaryKey>
                            </primaryKeys>
                        </syntheticObjects>
                    </database>
                </generator>
            </configuration>
            """, org.jooq.meta.jaxb.Configuration.class)));
```

The output is:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<configuration>
    <onError>FAIL</onError>
    <onUnused>LOG</onUnused>
    <generator>
        <name>org.jooq.codegen.DefaultGenerator</name>
        <database>
            <regexMatchesPartialQualification>true</regexMatchesPartialQualification>
            <sqlMatchesPartialQualification>true</sqlMatchesPartialQualification>
            <includeExcludeColumns>false</includeExcludeColumns>
            <includeExcludePackageRoutines>false</includeExcludePackageRoutines>
            <includeTables>true</includeTables>
            <includeEmbeddables>true</includeEmbeddables>
            <includeRoutines>true</includeRoutines>
            <includeTriggerRoutines>false</includeTriggerRoutines>
            <includePackages>true</includePackages>
            <includePackageRoutines>true</includePackageRoutines>
            <includePackageUDTs>true</includePackageUDTs>
            <includePackageConstants>true</includePackageConstants>
            <includeXMLSchemaCollections>true</includeXMLSchemaCollections>
            <includeUDTs>true</includeUDTs>
            <includeDomains>true</includeDomains>
            <includeSequences>true</includeSequences>
            <includeIndexes>true</includeIndexes>
            <includePrimaryKeys>true</includePrimaryKeys>
            <includeUniqueKeys>true</includeUniqueKeys>
            <includeForeignKeys>true</includeForeignKeys>
            <includeCheckConstraints>true</includeCheckConstraints>
            <includeSystemTables>false</includeSystemTables>
            <includeSystemIndexes>false</includeSystemIndexes>
            <includeSystemCheckConstraints>false</includeSystemCheckConstraints>
            <includeSystemSequences>false</includeSystemSequences>
            <includeSystemUDTs>false</includeSystemUDTs>
            <includeInvisibleColumns>true</includeInvisibleColumns>
            <recordVersionFields></recordVersionFields>
            <recordTimestampFields></recordTimestampFields>
            <syntheticObjects>
                <primaryKeys>
                    <primaryKey>
                        <name>AKSEPTANSETYPE__PK</name>
                        <tables>KJERNEAPI\.AKSEPTANSETYPE</tables>
                        <fields>
                            <field></field> <!-- Missing content here -->
                            <field></field>
                        </fields>
                    </primaryKey>
                </primaryKeys>
            </syntheticObjects>
            <syntheticIdentities></syntheticIdentities>
            <syntheticPrimaryKeys></syntheticPrimaryKeys>
            <overridePrimaryKeys></overridePrimaryKeys>
            <dateAsTimestamp>false</dateAsTimestamp>
            <ignoreProcedureReturnValues>false</ignoreProcedureReturnValues>
            <unsignedTypes>true</unsignedTypes>
            <integerDisplayWidths>true</integerDisplayWidths>
            <inputCatalog></inputCatalog>
            <outputCatalogToDefault>false</outputCatalogToDefault>
            <inputSchema></inputSchema>
            <outputSchemaToDefault>false</outputSchemaToDefault>
            <schemaVersionProvider></schemaVersionProvider>
            <catalogVersionProvider></catalogVersionProvider>
            <orderProvider></orderProvider>
            <readonlyIdentities>false</readonlyIdentities>
            <readonlyComputedColumns>true</readonlyComputedColumns>
            <readonlyNonUpdatableColumns>true</readonlyNonUpdatableColumns>
            <forcedTypesForBuiltinDataTypeExtensions>true</forcedTypesForBuiltinDataTypeExtensions>
            <forcedTypesForXMLSchemaCollections>true</forcedTypesForXMLSchemaCollections>
            <forceIntegerTypesOnZeroScaleDecimals>true</forceIntegerTypesOnZeroScaleDecimals>
            <logSlowQueriesAfterSeconds>5</logSlowQueriesAfterSeconds>
            <logSlowResultsAfterSeconds>5</logSlowResultsAfterSeconds>
        </database>
    </generator>
</configuration>
```

Will investigate right away

**Comment by lukaseder** [comment]
Ah yes, `MiniJAXB::unmarshalList0` can only unmarshal lists of `xs:complexType`, not lists of `xs:simpleType`. Should be an easy fix

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.4 (https://github.com/jOOQ/jOOQ/issues/13899), and 3.16.10 (https://github.com/jOOQ/jOOQ/issues/13900)

**Comment by alf** [comment]
Sweet, thank you for the quick turnaround. I will test the snapshots tomorrow.

**Comment by lukaseder** [comment]
Note, a workaround could be to put an actual JAXB implementation on the classpath, so that would be used instead of `MiniJAXB`

**Comment by alf** [comment]
> Note, a workaround could be to put an actual JAXB implementation on the classpath, so that would be used instead of `MiniJAXB`

I’ll try that today then. Good to know.

**Comment by alf** [comment]
Just a quick heads up that I can't get the workaround up and running. The way I read the code `Plugin.read` calls `GenerationTool.load` which calls the static method `MiniJAXB.unmarshal`. I don't understand how having a JAXB-implementation on the classpath could help. This isn't really important to me as I'll just upgrade to the fixed version, but I thought it might be of interest.

**Comment by lukaseder** [comment]
OK, my bad. I was under the impression that jOOQ still defaults to `JAXB` when it is on the classpath, and uses our own implementation only when `JAXB` isn't available. But in fact, that changed in jOOQ 3.12: https://github.com/jOOQ/jOOQ/issues/8914

I'm sorry about the wasted time. In that case, the upgrade is the only way.

**Comment by alf** [comment]
No problem. I didn’t waste much and I find it fun to dig into the jOOQ internals. 🤷‍♂️😅

**Comment by lukaseder** [comment]
Whew 😅

**Comment by alf** [comment]
Snapshot works as expected. Thanks.

**Comment by lukaseder** [comment]
3.16.10 and 3.17.4 have been released

**Comment by gpor0** [comment]
@lukaseder any chance this fix is backported to 3.15.x (pre-jakarta)? Thanks.

**Comment by lukaseder** [comment]
@gpor0 I'm no longer maintaining jOOQ 3.15. What's the reason you can't upgrade? Are you licensed (in case of which we might discuss things)

**Comment by gpor0** [comment]
After the investigation I see I can upgrade to Maria 10.6 since it is compatible with 3.16 and remove the Jakarta annotations since I don't use persistence validation so it should work for me.
